src/cryptsetup/cryptsetup-generator.c

```diff
@@ -110,7 +110,6 @@ static int create_disk(
                 "SourcePath=/etc/crypttab\n"
                 "DefaultDependencies=no\n"
                 "Conflicts=umount.target\n"
-                "BindsTo=dev-mapper-%i.device\n"
```

Comment:
You will get cosmetic display issue where `systemd-cryptsetup@.service` will be online but corresponding device - offline. This does mean that later `systemctl start` may become noop though. This sounds like we actually want `systemd-cryptsetup@.service` to be `PartOf=dev-mapper-%i.device`, except I am not sure whether `PartOf` also kicks in in case of unexpected removal.

As was demonstrated in issue 4756, `BindsTo` should really be "just that" - "unexpected removal condition", without overloading it with `Requires` semantic. Then it could be used as additional qualifier on all other dependencies - `Requires`, `Requisite` and `PartOf`, just extending "request to stop unit" with "unexpected unit removal".